<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>_systemPipeRdata_: Workflow templates and sample data • systemPipeRdata</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="_systemPipeRdata_: Workflow templates and sample data">
<meta property="og:description" content="systemPipeRdata">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">systemPipeRdata</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.19.9</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/systemPipeRdata.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/systemPipeChIPseq.html">ChIP-Seq Workflow Template</a>
    </li>
    <li>
      <a href="../articles/systemPipeRIBOseq.html">RIBO-Seq Workflow Template</a>
    </li>
    <li>
      <a href="../articles/systemPipeRNAseq.html">RNA-Seq Workflow Template</a>
    </li>
    <li>
      <a href="../articles/systemPipeVARseq.html">VAR-Seq Workflow Template</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/tgirke/systemPipeRdata/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="systemPipeRdata_files/header-attrs-2.7.12/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>
<em>systemPipeRdata</em>: Workflow templates and sample data</h1>
                        <h4 class="author">Author: Daniela Cassol (<a href="mailto:danielac@ucr.edu" class="email">danielac@ucr.edu</a>) and Thomas Girke (<a href="mailto:thomas.girke@ucr.edu" class="email">thomas.girke@ucr.edu</a>)</h4>
            
            <h4 class="date">Last update: 04 May, 2021</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/tgirke/systemPipeRdata/blob/master/vignettes/systemPipeRdata.Rmd"><code>vignettes/systemPipeRdata.Rmd</code></a></small>
      <div class="hidden name"><code>systemPipeRdata.Rmd</code></div>

    </div>

    
    
<style type="text/css">
pre code {
white-space: pre !important;
overflow-x: scroll !important;
word-break: keep-all !important;
word-wrap: initial !important;
}
</style>
<!---
- Compile from command-line
Rscript -e "rmarkdown::render('systemPipeRdata.Rmd', c('BiocStyle::html_document'), clean=F); knitr::knit('systemPipeRdata.Rmd', tangle=TRUE)"; Rscript -e "rmarkdown::render('systemPipeRdata.Rmd', c('BiocStyle::pdf_document'))"

- Commit to github
git commit -am "some edits"; git push -u origin master
--><script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  document.querySelector("h1").className = "title";
});
</script><script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  var links = document.links;  
  for (var i = 0, linksLength = links.length; i < linksLength; i++)
    if (links[i].hostname != window.location.hostname)
      links[i].target = '_blank';
});
</script><p><strong>Note:</strong> the most recent version of this vignette can be found <a href="https://github.com/tgirke/systemPipeRdata/blob/master/vignettes/systemPipeRdata.Rmd">here</a>.</p>
<p><strong>Note:</strong> if you use <em><code>systemPipeR</code></em> and <em><code>systemPipeRdata</code></em> in published research, please cite:</p>
<p>Backman, T.W.H and Girke, T. (2016). <em>systemPipeR</em>: Workflow and Report Generation Environment. <em>BMC Bioinformatics</em>, 17: 388. <a href="https://doi.org/10.1186/s12859-016-1241-0">10.1186/s12859-016-1241-0</a>.</p>
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p><a href="https://github.com/tgirke/systemPipeRdata"><em><code>systemPipeRdata</code></em></a> is a helper package to generate with a single command workflow templates that are intended to be used by its parent package <a href="http://www.bioconductor.org/packages/devel/bioc/html/systemPipeR.html"><em><code>systemPipeR</code></em></a> <span class="citation">(H Backman and Girke 2016)</span>. The <em>systemPipeR</em> project provides a suite of R/Bioconductor packages for designing, building and running end-to-end analysis workflows on local machines, HPC clusters and cloud systems, while generating at the same time publication quality analysis reports.</p>
<p>To test workflows quickly or design new ones from existing templates, users can generate with a single command workflow instances fully populated with sample data and parameter files required for running a chosen workflow. Pre-configured directory structure of the workflow environment and the sample data used by <em><code>systemPipeRdata</code></em> are described <a href="http://bioconductor.org/packages/release/bioc/vignettes/systemPipeR/inst/doc/systemPipeR.html#load-sample-data-and-workflow-templates">here</a>.</p>
</div>
<div id="getting-started" class="section level1">
<h1 class="hasAnchor">
<a href="#getting-started" class="anchor"></a>Getting started</h1>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<p>The <em><code>systemPipeRdata</code></em> package is available at <a href="http://www.bioconductor.org/packages/release/data/experiment/html/systemPipeRdata.html">Bioconductor</a> and can be installed from within R as follows:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span>
<span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span><span class="op">(</span><span class="st">"systemPipeRdata"</span><span class="op">)</span></code></pre></div>
<p>Also, it is possible to install the development version from <a href="http://www.bioconductor.org/packages/devel/data/experiment/html/systemPipeRdata.html">Bioconductor</a>.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span><span class="op">(</span><span class="st">"systemPipeRdata"</span>, version <span class="op">=</span> <span class="st">"devel"</span>, build_vignettes <span class="op">=</span> <span class="cn">TRUE</span>, 
    dependencies <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>  <span class="co"># Installs Devel version from Bioconductor</span></code></pre></div>
</div>
<div id="loading-package-and-documentation" class="section level2">
<h2 class="hasAnchor">
<a href="#loading-package-and-documentation" class="anchor"></a>Loading package and documentation</h2>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/tgirke/systemPipeRdata">"systemPipeRdata"</a></span><span class="op">)</span>  <span class="co"># Loads the package</span></code></pre></div>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span>help <span class="op">=</span> <span class="st">"systemPipeRdata"</span><span class="op">)</span>  <span class="co"># Lists package info</span>
<span class="fu"><a href="https://rdrr.io/r/utils/vignette.html">vignette</a></span><span class="op">(</span><span class="st">"systemPipeRdata"</span><span class="op">)</span>  <span class="co"># Opens vignette</span></code></pre></div>
</div>
<div id="starting-with-pre-configured-workflow-templates" class="section level2">
<h2 class="hasAnchor">
<a href="#starting-with-pre-configured-workflow-templates" class="anchor"></a>Starting with pre-configured workflow templates</h2>
<p>Load one of the available workflows into your current working directory. The following does this for the <em><code>varseq</code></em> workflow template. The name of the resulting workflow directory can be specified under the <em><code>mydirname</code></em> argument. The default <em><code>NULL</code></em> uses the name of the chosen workflow. An error is issued if a directory of the same name and path exists already.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/genWorkenvir.html">genWorkenvir</a></span><span class="op">(</span>workflow <span class="op">=</span> <span class="st">"systemPipeR/SPvarseq"</span>, mydirname <span class="op">=</span> <span class="st">"varseq"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/getwd.html">setwd</a></span><span class="op">(</span><span class="st">"varseq"</span><span class="op">)</span></code></pre></div>
<p>On Linux and OS X systems the same can be achieved from the command-line of a terminal with the following commands.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="ex">$</span> Rscript <span class="at">-e</span> <span class="st">"systemPipeRdata::genWorkenvir(workflow='systemPipeR/SPvarseq', mydirname='varseq')"</span></span></code></pre></div>
</div>
</div>
<div id="workflow-templates-collection" class="section level1">
<h1 class="hasAnchor">
<a href="#workflow-templates-collection" class="anchor"></a>Workflow templates collection</h1>
<p>A collection of workflow templates are available, and it is possible to browse the current availability, as follows:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/availableWF.html">availableWF</a></span><span class="op">(</span>github <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p>This function returns the list of workflow templates available within the package and <a href="https://github.com/systemPipeR">systemPipeR Organization</a> on GitHub. Each one listed template can be created as described above.</p>
<p>The workflow template choose from Github will be installed as an R package, and also it creates the environment with all the settings and files to run the demo analysis.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/genWorkenvir.html">genWorkenvir</a></span><span class="op">(</span>workflow<span class="op">=</span><span class="st">"systemPipeR/SPrnaseq"</span>, mydirname<span class="op">=</span><span class="st">"NULL"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/getwd.html">setwd</a></span><span class="op">(</span><span class="st">"SPrnaseq"</span><span class="op">)</span></code></pre></div>
<p>Besides, it is possible to choose different versions of the workflow template, defined through other branches on the GitHub Repository. By default, the <em><code>master</code></em> branch is selected, however, it is possible to define a different branch with the <em><code>ref</code></em> argument.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/genWorkenvir.html">genWorkenvir</a></span><span class="op">(</span>workflow<span class="op">=</span><span class="st">"systemPipeR/SPrnaseq"</span>, ref <span class="op">=</span> <span class="st">"singleMachine"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/getwd.html">setwd</a></span><span class="op">(</span><span class="st">"SPrnaseq"</span><span class="op">)</span></code></pre></div>
<div id="download-a-specific-r-markdown-file" class="section level2">
<h2 class="hasAnchor">
<a href="#download-a-specific-r-markdown-file" class="anchor"></a>Download a specific R Markdown file</h2>
<p>Also, it is possible to download a specific workflow script for your analysis. The URL can be specified under <em><code>url</code></em> argument and the R Markdown file name in the <em><code>urlname</code></em> argument. The default <em><code>NULL</code></em> copies the current version available in the chose template.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/genWorkenvir.html">genWorkenvir</a></span><span class="op">(</span>workflow<span class="op">=</span><span class="st">"systemPipeR/SPrnaseq"</span>, url <span class="op">=</span> <span class="st">"https://raw.githubusercontent.com/systemPipeR/systemPipeRNAseq/cluster/vignettes/systemPipeRNAseq.Rmd"</span>, 
             urlname <span class="op">=</span> <span class="st">"rnaseq_V-cluster.Rmd"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/getwd.html">setwd</a></span><span class="op">(</span><span class="st">"rnaseq"</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div id="dynamic-generation-of-workflow-template" class="section level1">
<h1 class="hasAnchor">
<a href="#dynamic-generation-of-workflow-template" class="anchor"></a>Dynamic generation of workflow template</h1>
<p>It is possible to create a new workflow structure from RStudio menu <code>File -&gt; New File -&gt; R Markdown -&gt; From Template -&gt; systemPipeR New WorkFlow</code>. This interactive option creates the same environment as demonstrated above.</p>
<p><img src="results/rstudio.png"><strong>Figure 1:</strong> Selecting workflow template within RStudio.</p>
</div>
<div id="directory-structure" class="section level1">
<h1 class="hasAnchor">
<a href="#directory-structure" class="anchor"></a>Directory Structure</h1>
<p>The workflow templates generated by <em><code>genWorkenvir</code></em> contain the following preconfigured directory structure:</p>
<ul>
<li>
<span style="color:green"><em><strong>workflow/</strong></em></span> (<em>e.g.</em> <em>rnaseq/</em>)
<ul>
<li>This is the root directory of the R session running the workflow.</li>
<li>Run script ( <em>*.Rmd</em>) and sample annotation (<em>targets.txt</em>) files are located here.</li>
<li>Note, this directory can have any name (<em>e.g.</em> <span style="color:green"><em><strong>rnaseq</strong></em></span>, <span style="color:green"><em><strong>varseq</strong></em></span>). Changing its name does not require any modifications in the run script(s).</li>
<li>
<strong>Important subdirectories</strong>:
<ul>
<li>
<span style="color:green"><em><strong>param/</strong></em></span>
<ul>
<li>Stores non-CWL parameter files such as: <em>*.param</em>, <em>*.tmpl</em> and <em>*.run.sh</em>. These files are only required for backwards compatibility to run old workflows using the previous custom command-line interface.</li>
<li>
<span style="color:green"><em><strong>param/cwl/</strong></em></span>: This subdirectory stores all the CWL parameter files. To organize workflows, each can have its own subdirectory, where all <code>CWL param</code> and <code>input.yml</code> files need to be in the same subdirectory.</li>
</ul>
</li>
<li>
<span style="color:green"><em><strong>data/</strong></em> </span>
<ul>
<li>FASTQ files</li>
<li>FASTA file of reference (<em>e.g.</em> reference genome)</li>
<li>Annotation files</li>
<li>etc.</li>
</ul>
</li>
<li>
<span style="color:green"><em><strong>results/</strong></em></span>
<ul>
<li>Analysis results are usually written to this directory, including: alignment, variant and peak files (BAM, VCF, BED); tabular result files; and image/plot files</li>
<li>Note, the user has the option to organize results files for a given sample and analysis step in a separate subdirectory.</li>
</ul>
</li>
</ul>
</li>
</ul>
</li>
</ul>
<p><strong>Note</strong>: Directory names are indicated in <span style="color:grey"><strong><em>green</em></strong></span>. Users can change this structure as needed, but need to adjust the code in their workflows accordingly.</p>
<center>
<img src="results/directory.png">
</center>
<p><strong>Figure 2:</strong> <em>systemPipeR’s</em> preconfigured directory structure.</p>
</div>
<div id="run-workflows" class="section level1">
<h1 class="hasAnchor">
<a href="#run-workflows" class="anchor"></a>Run workflows</h1>
<p>Next, run from within R the chosen sample workflow by executing the code provided in the corresponding <em><code>*.Rmd</code></em> template file. Much more detailed information on running and customizing <a href="http://www.bioconductor.org/packages/devel/bioc/html/systemPipeR.html"><em><code>systemPipeR</code></em></a> workflows is available in its overview vignette <a href="http://www.bioconductor.org/packages/devel/bioc/vignettes/systemPipeR/inst/doc/systemPipeR.html">here</a>. This vignette can also be opened from R with the following command.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st"><a href="https://systempipe.org/">"systemPipeR"</a></span><span class="op">)</span>  <span class="co"># Loads systemPipeR which needs to be installed via BiocManager::install() from Bioconductor</span></code></pre></div>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/vignette.html">vignette</a></span><span class="op">(</span><span class="st">"systemPipeR"</span>, package <span class="op">=</span> <span class="st">"systemPipeR"</span><span class="op">)</span></code></pre></div>
<div id="return-paths-to-sample-data" class="section level2">
<h2 class="hasAnchor">
<a href="#return-paths-to-sample-data" class="anchor"></a>Return paths to sample data</h2>
<p>The location of the sample data provided by <em><code>systemPipeRdata</code></em> can be returned as a <em><code>list</code></em>.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/pathList.html">pathList</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## $targets
## [1] "/tmp/Rtmp6btZ98/temp_libpath12422253345dc/systemPipeRdata/extdata/param/targets.txt"
## 
## $targetsPE
## [1] "/tmp/Rtmp6btZ98/temp_libpath12422253345dc/systemPipeRdata/extdata/param/targetsPE.txt"
## 
## $annotationdir
## [1] "/tmp/Rtmp6btZ98/temp_libpath12422253345dc/systemPipeRdata/extdata/annotation/"
## 
## $fastqdir
## [1] "/tmp/Rtmp6btZ98/temp_libpath12422253345dc/systemPipeRdata/extdata/fastq/"
## 
## $bamdir
## [1] "/tmp/Rtmp6btZ98/temp_libpath12422253345dc/systemPipeRdata/extdata/bam/"
## 
## $paramdir
## [1] "/tmp/Rtmp6btZ98/temp_libpath12422253345dc/systemPipeRdata/extdata/param/"
## 
## $workflows
## [1] "/tmp/Rtmp6btZ98/temp_libpath12422253345dc/systemPipeRdata/extdata/workflows/"
## 
## $chipseq
## [1] "/tmp/Rtmp6btZ98/temp_libpath12422253345dc/systemPipeRdata/extdata/workflows/chipseq/"
## 
## $rnaseq
## [1] "/tmp/Rtmp6btZ98/temp_libpath12422253345dc/systemPipeRdata/extdata/workflows/rnaseq/"
## 
## $riboseq
## [1] "/tmp/Rtmp6btZ98/temp_libpath12422253345dc/systemPipeRdata/extdata/workflows/riboseq/"
## 
## $varseq
## [1] "/tmp/Rtmp6btZ98/temp_libpath12422253345dc/systemPipeRdata/extdata/workflows/varseq/"
## 
## $new
## [1] "/tmp/Rtmp6btZ98/temp_libpath12422253345dc/systemPipeRdata/extdata/workflows/new/"</code></pre>
</div>
</div>
<div id="version-information" class="section level1">
<h1 class="hasAnchor">
<a href="#version-information" class="anchor"></a>Version information</h1>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## R Under development (unstable) (2021-02-04 r79940)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 20.04.2 LTS
## 
## Matrix products: default
## BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.9.0
## LAPACK: /home/dcassol/src/R-devel/lib/libRlapack.so
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets 
## [6] methods   base     
## 
## other attached packages:
## [1] systemPipeRdata_1.19.9 BiocStyle_2.19.2      
## 
## loaded via a namespace (and not attached):
##  [1] knitr_1.33          magrittr_2.0.1     
##  [3] BiocGenerics_0.37.4 R6_2.5.0           
##  [5] ragg_1.1.2          rlang_0.4.10       
##  [7] fastmap_1.1.0       stringr_1.4.0      
##  [9] tools_4.1.0         parallel_4.1.0     
## [11] xfun_0.22           jquerylib_0.1.4    
## [13] remotes_2.3.0       htmltools_0.5.1.1  
## [15] systemfonts_1.0.1   yaml_2.2.1         
## [17] digest_0.6.27       rprojroot_2.0.2    
## [19] pkgdown_1.6.1       crayon_1.4.1       
## [21] bookdown_0.22       textshaping_0.3.3  
## [23] formatR_1.9         BiocManager_1.30.12
## [25] codetools_0.2-18    sass_0.3.1         
## [27] fs_1.5.0            memoise_2.0.0      
## [29] cachem_1.0.4        evaluate_0.14      
## [31] rmarkdown_2.7.12    stringi_1.5.3      
## [33] compiler_4.1.0      bslib_0.2.4        
## [35] desc_1.3.0          jsonlite_1.7.2</code></pre>
</div>
<div id="funding" class="section level1">
<h1 class="hasAnchor">
<a href="#funding" class="anchor"></a>Funding</h1>
<p>This project was supported by funds from the National Institutes of Health (NIH) and the National Science Foundation (NSF).</p>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-H_Backman2016-bt" class="csl-entry">
H Backman, Tyler W, and Thomas Girke. 2016. <span>“<span class="nocase">systemPipeR: NGS workflow and report generation environment</span>.”</span> <em>BMC Bioinformatics</em> 17 (1): 388. <a href="https://doi.org/10.1186/s12859-016-1241-0">https://doi.org/10.1186/s12859-016-1241-0</a>.
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by <a href="http://girke.bioinformatics.ucr.edu/">Thomas Girke</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
